### PR TITLE
Bypass Usage of Drupal Constants in Content-Build

### PR DESCRIPTION
**Pull request by olivereri**
## Description
Due to security requirements CMS will be disabling the shared Content API Credentials `api`. If CLI arguments or environment variables aren't supplied for drupal-user or drupal-password the content-build will use hard coded credentials in drupal.js constants file. The hard coded credentials are for the `api` user and will no longer be valid. Developers and other automation will have to obtain and use specific users to send requests to the Drupal API.

## Relations
https://github.com/department-of-veterans-affairs/va.gov-cms/issues/7965

## Testing done
- local `yarn build --pull-drupal`
- All Continuous Integration tests have passed.

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Comment by rmessina1010** [comment]
CI fail...
`Missing --drupal-user CLI argument or DRUPAL_USERNAME environment variable not set.`

**Comment by rmessina1010** [comment]
Still failing, could it be a credential issue?? ```
If you do not have Drupal API credentials please request them from:
#cms-support https://dsva.slack.com/archives/CDHBKAL9W
```

**Comment by olivereri** [comment]
> Still failing, could it be a credential issue?? ``` If you do not have Drupal API credentials please request them from: #cms-support https://dsva.slack.com/archives/CDHBKAL9W

This is failing because, all of the workflows that depend on `yarn build` haven't been updated to explicitly include `--drupal-user` and `--drupal-password` by CLI or Env var. The point of this PR is remove stored credentials in the content-build app and force the use of CLI args or Env vars.

Once the other workflows are updated this will pass tests.

**Review by timwright12 — APPROVED** [review]
This is fine, but please do make sure you announce this update to VFS teams

**Comment by olivereri** [comment]
> This is fine, but please do make sure you announce this update to VFS teams

https://dsva.slack.com/archives/CE4304QPK/p1645646821894319

File: README.md [hunk 1]
```diff
@@ -25,7 +25,39 @@ Once you have the site set up locally, these are some common commands you might
...
+- example file content:
+```
+#.env file
+DRUPAL_ADDRESS=https://content-build-medc0xjkxm4jmpzxl3tfbcs7qcddsivh.ci.cms.va.gov
+DRUPAL_USERNAME=content_build_api
+DRUPAL_PASSWORD=drupal8
+```
+In most cases the default environment variables in `.env` will be all that's necessary. 
+If you do however require Production credentials to pull content from https://prod.cms.va.gov please contact 
+[#cms-support](https://dsva.slack.com/archives/CDHBKAL9W) and request user creation with permissions to read the Drupal Content API.
```
Comment (timcosgrove) [inline README.md:57]: I tried rewriting this to emphasize a) the env.example is probably all they're going to need and b) they can override at the command line if they need to. If you don't like my changes, discard, I won't take offense :)

Comment (timcosgrove) [reply README.md:57]: Authentication credentials are stored in a local .gitignore'd file `.env`. An example file is provided to get you started: 

`cp .env.example .env` 

File contents:
```
# .env file contents
DRUPAL_ADDRESS=https://content-build-medc0xjkxm4jmpzxl3tfbcs7qcddsivh.ci.cms.va.gov
DRUPAL_USERNAME=content_build_api
DRUPAL_PASSWORD=drupal8
```

In most cases, these values will be all you need to run a content build with fresh CMS content. However, if you need to connect to a different CMS endpoint, you can either override the values in the `.env` file, or else override specific values at the command line. For example:

`yarn build --pull-drupal --drupal-address=https://some-other-endpoint.cms.va.gov`

If you do need to pull content from https://prod.cms.va.gov please contact 
[#cms-support](https://dsva.slack.com/archives/CDHBKAL9W) and request a user with permissions to read the Drupal Content API. You will need to use your assigned credentials to access Production.

**Review by timcosgrove — APPROVED** [review]
+r

I added some language suggestions but if you don't like them feel free to discard them.

File: README.md (hunk 1 above)
Comment (olivereri) [reply README.md]: I like it!

**Comment by timwright12** [comment]
@olivereri Do we also need updates to the content-release and daily-production-deploy workflows for this? They won't likely have those credentials

**Comment by olivereri** [comment]
> I think we need a little more testing here. Can we run a branch off this through CI to make sure the user/pass are working correctly? This feels like one of those things that will cause mayhem if it goes wrong

@timwright12 I disagree, this PR  https://github.com/department-of-veterans-affairs/content-build/pull/980 covered that and was thoroughly tested. The user/pass updates have been running since 12pm ET yesterday. This current PR just removes Drupal API connection information from javascript, does error handling in case that connection information isn't supplied, and updates documentation and provides example connection information. Finally, Integration tests for this branch passed after #980 was merged.

**Comment by timwright12** [comment]
@olivereri can you list that stuff in the "Testing done" section? It just says local right now, which really isn't enough for a change like this

**Comment by timwright12** [comment]
@olivereri We usually also either run through a full test plan or at the very least download the assets generated in CI and run them to make sure the site is still being build correctly and the data is coming in the way we expect it

**Review by timwright12 — APPROVED** [review]

**Pull request merged**